processing/src/main/java/io/druid/segment/IndexMergerV9.java

```diff
@@ -644,7 +644,7 @@ private void mergeIndexesAndWriteColumns(
         if (dimensionSkipFlag.get(i)) {
           continue;
         }
-        if (dims[i] == null || dims[i].length == 0) {
+        if (dims[i] == null || dims[i].length == 0 || dims[i][0] == 0) {
```

Comment:
Is dim val 0 guaranteed to be a null/emptystr?

Even if it is, this last check should be `dims[i].length == 1 && dims[i][0] == 0` as the dim value could potentially be an empty str + something else.